src/components/BBCode/BBCode.tsx

```diff
@@ -1,15 +1,77 @@
-import React, { FC } from 'react';
-
-import BBCodeUntyped from '@bbob/react/es/Component';
-import reactPreset from '@bbob/preset-react/es';
-
-import cls from './BBCode.module.css';
-
-/**
- * Typed BBCode component.
- */
-export const BBCode: FC = ({ children }) => (
-  <BBCodeUntyped plugins={[reactPreset()]} componentProps={{ className: cls.bbcode }}>
-    {children}
-  </BBCodeUntyped>
-);
+import React, { ReactElement, VFC } from 'react';
+
+import { parse, ast, ast_item, ast_to_array } from '@prekel/rescript-bbcode/src/BBCode.gen';
+import { Box, Link, Text } from '@chakra-ui/react';
+import cls from './BBCode.module.css';
+
+/**
+ * Transforms AST to React element array through astItemToElement function.
+ * @param parsed Parsed representation of BBCode-text (AST).
+ */
+const mapAstToElements = (parsed: ast): ReactElement[] => {
+  /**
+   * Transforms AST item to corresponding React element.
+   * @param astItem An AST item.
+   * @param index Index which used in key.
+   */
+  const mapAstItemToElement = (astItem: ast_item, index: number): ReactElement => {
+    const itemKey = `${astItem.tag}${index.toString()}`;
+    switch (astItem.tag) {
+      case 'Text': return <Text key={itemKey} as="span" className={cls.text}>{astItem.value}</Text>;
+      case 'Bold': return <Text key={itemKey} as="b">{mapAstToElements(astItem.value.children)}</Text>;
+      case 'Italic': return <Text key={itemKey} as="i">{mapAstToElements(astItem.value.children)}</Text>;
+      case 'Underline': return <Text key={itemKey} as="u">{mapAstToElements(astItem.value.children)}</Text>;
+      case 'Strikethrough': return <Text key={itemKey} as="s">{mapAstToElements(astItem.value.children)}</Text>;
+      case 'Link': return <Link key={itemKey} href={astItem.value.url} isExternal />;
+      case 'LinkNamed':
+        // TODO: Handle vndb-relative url (like "/v3126").
+        return <Link key={itemKey} href={astItem.value.url} isExternal>{mapAstToElements(astItem.value.children)}</Link>;
+      case 'Spoiler':
+        // TODO: Display spoiler via Chakra components.

```

Comment:
So, are you planning to close these TODOs in this task? Or not?

Reply:
Btw, don't forget to add your identity to TODO comments, like this: `// TODO (Prekel V.): Handle...`